sw/host/opentitanlib/src/io/i2c.rs

```diff
@@ -10,24 +10,40 @@ use thiserror::Error;
 
 use crate::app::TransportWrapper;
 use crate::impl_serializable_error;
+use crate::util::parse_int::ParseInt;
 
 #[derive(Debug, Args)]
 pub struct I2cParams {
     /// I2C instance.
-    #[arg(long, default_value = "0")]
-    pub bus: String,
+    #[arg(long)]
+    pub bus: Option<String>,
 
     /// I2C bus speed (typically: 100000, 400000, 1000000).
     #[arg(long)]
     pub speed: Option<u32>,
+
+    /// 7 bit I2C device address.
+    #[arg(
+        short,
+        long,
+        value_parser = u8::from_str
+    )]
+    pub addr: Option<u8>,
 }
 
 impl I2cParams {
-    pub fn create(&self, transport: &TransportWrapper) -> Result<Rc<dyn Bus>> {
-        let i2c = transport.i2c(&self.bus)?;
+    pub fn create(
+        &self,
+        transport: &TransportWrapper,
+        default_instance: &str,
```

Comment:
That is correct, the `default_instance` is a hard-coded "default" of each command or other place where an I2C object is instantiated based on command line switches.  E.g. `opentitantool i2c tpm ...` will by default use the I2C target called "TPM", and so on.  This is how SPI already works.

You are right that there could be commands (`spi raw-...` and `i2c raw-...` for instance) for which there is no generally useful default, and it would be better if these commands required an explicit bus being specified.